staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -89,6 +89,12 @@ type Config struct {
 	// Exec-based authentication provider.
 	ExecProvider *clientcmdapi.ExecConfig
 
+	// ExecStdinUnavailable indicates whether the above ExecProvider can pass standard
+	// input through to the exec plugin. For example, a higher level entity might be using
+	// standard input for something else and therefore it would not be safe for the exec
+	// plugin to use standard input.
+	ExecStdinUnavailable bool
```

Comment:
Should this be inside `ExecConfig`?  Similar to `LocationOfOrigin`?

```go
// Cluster contains information about how to communicate with a kubernetes cluster
type Cluster struct {
	// LocationOfOrigin indicates where this object came from.  It is used for round tripping config post-merge, but never serialized.
	// +k8s:conversion-gen=false
	LocationOfOrigin string
...
}

Reply:
I'm glad you brought this up. I didn't put it in `ExecConfig` because I didn't think we wanted a user to be able to set it in their kubeconfig. But I do agree this is weird to have it hanging off `rest.Config`. It looks like `LocationOfOrigin` follows the pattern of "this field should logically be on the kubeconfig struct, but it shouldn't be serialized".

Gave this a go here: e91ed5ce232cdd45d0e3bc719e0e5ee3454601bb.